feat(dev-toolbar): Dev Toolbar Docs (#12117)
getsentry/sentry-docs: docs/product/dev-toolbar/index.mdx

---

> title: Dev Toolbar

Author: we want to use Dev Toolbar I think cc @jas-kas

> title: Dev Toolbar

Reviewer: I believe we settled on Dev Toolbar moving forward

[Learn more](/product/explore/feature-flags/) about what data Sentry can track related to your feature flagging system.

![Dev Toolbar with the Feature Flags Panel open](./img/feature-flags-panel.png)

## Login Flow

After setting up the Dev Toolbar you’ll see a “Login to Sentry” button floating in the center of the page. Your team will need to click this button and login to sentry before they can access any information from your Sentry organization.
![Dev Toolbar login button](./img/login-button.png)

After clicking the button, you should see a pop up for logging in to Sentry.
![Sentry log in pop up](./img/login-popup.png)

If it's your first time setting up the Dev Toolbar, there will be a button with a link to Sentry's Dev Toolbar settings page to configure your domain.
![Dev Toolbar configure domains button](./img/configure-domain.png)

You'll need to add your domain URL to Allowed Origins on Sentry's Dev Toolbar settings page.
![Sentry's Dev Toolbar Settings Page](./img/sentry-toolbar-settings.png)

> Learn more about [deploying the toolbar to different environments](/product/dev-toolbar/setup/#deploying-to-production--dev-environments), and [conditionally inserting the toolbar script tag](/product/dev-toolbar/setup/#conditionally-inserting-script-tag).

Reviewer 2: Lets add a link to the sub page from here?

```suggestion
Learn more about [deploying the toolbar to different environments](/product/dev-toolbar/setup/#deploying-to-production--dev-environments), and [conditionally inserting the toolbar script tag](/product/dev-toolbar/setup/#conditionally-inserting-script-tag).

[Set Up](..)
```

---
title: Dev Toolbar
sidebar_order: 120
description: "Bring critical Sentry insights and tools directly into your web app for easier troubleshooting with the Dev Toolbar."
---

<Alert level="info">
  The Dev Toolbar is currently in **beta**. Beta features are still in progress and may have bugs. Please reach out on
  [GitHub](https://github.com/getsentry/sentry-toolbar/issues) if you have any feedback or concerns.
</Alert>

Developers spend a lot of time troubleshooting their web apps on local, staging, and production environments. The Sentry Dev Toolbar pulls data from [sentry.io](http://sentry.io) and surfaces relevant actionable issues to you when you have the most context for understanding it: as you browse your own site. The Dev Toolbar is a floating widget in your web app, offering meaningful Sentry insights for the specific page being viewed through three different **page-aware** panels: Issues, Feedback, and Feature Flags.

> ![Collapsed Dev Toolbar in context of a website](./img/toolbar-collapsed.png)

Reviewer: I would say if possible this could even be a gif instead, so that the idea is to emphasize that it shouldn't get in the way of you browsing your site, but when you hover over it opens up (or is it when you click?)

> ![Collapsed Dev Toolbar in context of a website](./img/toolbar-collapsed.png)

Author: It opens on hover! unfortunately I don't know how to make a gif

## Issues Panel

> The issues panel shows you your highest priority frontend [issues](https://docs.sentry.io/product/issues/) for the page you are currently viewing. From the Dev Toolbar, you can quickly jump into specific issues in Sentry to get more detailed information and take action (for example, assign to a team member or mark as resolved). With the issues panel, you can browse the pages that are most important to your business and understand the top issues impacting your user base.

Reviewer: We also aren't mentioning here that just clicking on the Issue title will take you to your Issue _in Sentry_, I think that is important to call out here.

> The issues panel shows you your highest priority frontend [issues](https://docs.sentry.io/product/issues/) for the page you are currently viewing. From the Dev Toolbar, you can quickly jump into specific issues in Sentry to get more detailed information (e.g. Sentry project, issue title, and description, and when the issue was first and last seen) and take action (e.g. assign to a team member or mark as resolved). With the issues panel, you can browse the pages that are most important to your business and understand the top issues impacting your user base.

Reviewer 3: ```suggestion
The issues panel shows you your highest priority frontend [issues](https://docs.sentry.io/product/issues/) for the page you are currently viewing. From the Dev Toolbar, you can quickly jump into specific issues in Sentry to get more detailed information (examples: Sentry project, issue title, description, and when the issue was first and last seen) and take action (example: assign to a team member or mark as resolved). With the issues panel, you can browse the pages that are most important to your business and understand the top issues impacting your user base.
```
We try and avoid using latin phrases like 'eg' in the docs 👽

![Dev Toolbar with the Issues Panel open](./img/issues-panel.png)

## Feedback Panel

> The feedback panel shows you the most recent user feedback messages for the page you are on, so you can more easily contextualize user feedback. The feedback often describes suboptimal experiences that are not strictly code-thrown errors, such as misleading UX, broken links, and typos. The feedback panel works by pulling feedback messages from [Sentry’s User Feedback Widget](https://docs.sentry.io/product/user-feedback/#user-feedback-widget). Just like with the Issues panel, clicking on a particular user feedback will take you to the full detailed User Feedback page within Sentry.

Reviewer 3: ```suggestion
The feedback panel shows you the most recent user feedback messages for the page you are on, so you can more easily contextualize it. The feedback often describes suboptimal experiences that are not strictly code-thrown errors, such as misleading UX, broken links, and typos. The feedback panel works by pulling feedback messages from [Sentry’s User Feedback Widget](https://docs.sentry.io/product/user-feedback/#user-feedback-widget). Just like with the Issues panel, clicking on a particular user feedback will take you to the full detailed User Feedback page within Sentry.
```

![Dev Toolbar with the Feedback Panel open](./img/feedback-panel.png)

## Feature Flags Panel

> Feature flags are a powerful tool that allow you to control the visibility of features in your app, enabling you to ship, test, and experiment with confidence. The feature flag panel allows you to quickly view and override feature flags locally. You can override any feature flag to be `true` or `false` for your browser session, so you can verify its behavior and observe the impact it might have on errors. Learn more about how to [configure the feature flag panel](/product/dev-toolbar/setup/#implement-feature-flag-adapter).

Reviewer 3: ```suggestion
Feature flags are a powerful tool that allow you to control the visibility of features in your app, enabling you to ship, test, and experiment with confidence. The feature flag panel allows you to quickly view and override feature flags locally. You can override any feature flag to be `true` or `false` for your browser session, so you can verify its behavior and observe the impact it might have on errors. Learn more about how to [configure the feature flag panel](/product/dev-toolbar/setup/#implement-feature-flag-adapter) and [what data Sentry can track](/product/explore/feature-flags/) related to your feature flagging system.
```

Reviewer 3: I think including this link here is less disruptive to the flow of the doc, vs putting it in the note below.

> <Note>

Reviewer 3: ```suggestion

```

>     [Learn more](/product/explore/feature-flags/) about what data Sentry can track related to your feature flagging system.

Reviewer 3: ```suggestion
```

Reviewer 4: @coolguyzone why remove this?

> </Note>

Reviewer 3: ```suggestion
```

![Dev Toolbar with the Feature Flags Panel open](./img/feature-flags-panel.png)

## Login Flow

> After setting up the Dev Toolbar you’ll see a “Login to Sentry” button floating in the center of the page. To access any information from your Sentry organization you have to click this button and login to sentry.

Reviewer 3: ```suggestion
After setting up the Dev Toolbar you’ll see a “Login to Sentry” button floating in the center of the page. To access any information from your Sentry organization, you have to click this button and login to Sentry.
```

After clicking the button, you should see a pop up for logging in to Sentry.

> ![Sentry log in pop up](./img/login-popup.png)

Reviewer: Same comments about visibility and clarity of this image.

> After clicking the button, you will see a pop up for logging in to Sentry.

Reviewer 3: ```suggestion
After clicking the button, you will see a pop-up window for logging in to Sentry.
```

If it's your first time setting up the Dev Toolbar, there will be a button with a link to Sentry's Dev Toolbar settings page to configure your domain.

> ![Dev Toolbar configure domains button](./img/configure-domain.png)

Reviewer: Same comments about visibility and clarity of this image.

> You'll need to add your domain URL to Allowed Origins on Sentry's Dev Toolbar settings page.

Reviewer 4: ```suggestion
You'll need to add your domain URL to Allowed Origins on Sentry's Settings > Project > Dev Toolbar page.
```

something like that, i just want to mention that it's a Project specific settings page so people can find it. If it wasn't project specific i'd also suggest that we deep link to that page, but because it's project specific idk if we can make the link.

> ![Sentry's Dev Toolbar Settings Page](./img/sentry-toolbar-settings.png)

Reviewer: Same comments about visibility and clarity of this image.

> You'll need to add your domain URL to Allowed Origins in Sentry on the Settings > Project > Dev Toolbar page.

Reviewer 3: ```suggestion
You'll need to add your domain URL to Allowed Origins in Sentry on the **Settings > Project > Dev Toolbar** page.
```

> Learn more about [deploying the toolbar to different environments](/product/dev-toolbar/setup/#deploying-to-production--dev-environments), and [conditionally inserting the toolbar script tag](/product/dev-toolbar/setup/#conditionally-inserting-script-tag).

Reviewer 4: These links go to the bottom sections of the next page, which seems weird. Why not the top?

It looks like this, there is is a link "Next: Set Up"
![SCR-20241212-omak](https://github.com/user-attachments/assets/002fc3f1-b119-48ce-982e-84a1cb713b12)

I feel like, we should link people to the top of the next page, and use the sentence here as a teaser for what sections are on that page... 

> 

```suggestion
Learn more about [setting up the toolbar](/project/-dev-toolbar/setup/) including configuring a feature flag provider and including the toolbar in different environments.
```

maybe with all the links... but def the first link to the top of the page.

Reviewer: Do we have a reason to believe people don't click on "Next Set Up >>" and that's why we are adding the link to this sentence? I think we shouldn't have any link because then I would be wondering if the link in the sentence takes me to the same place as the button.

```suggestion
Learn more about setting up the toolbar including configuring a feature flag provider and including the toolbar in different environments on the next page.
```

> ### [Set Up](/product/dev-toolbar/setup)

Reviewer: ```suggestion

```
